Code/cmake/Modules/RDKitUtils.cmake

```diff
@@ -31,46 +36,46 @@ macro(rdkit_library)
     ${ARGN})
   CAR(RDKLIB_NAME ${RDKLIB_DEFAULT_ARGS})
   CDR(RDKLIB_SOURCES ${RDKLIB_DEFAULT_ARGS})
-  if(MSVC)
+  if(RDK_INSTALL_STATIC_LIBS AND NOT RDK_INSTALL_DYNAMIC_LIBS)
     add_library(${RDKLIB_NAME} ${RDKLIB_SOURCES})
     target_link_libraries(${RDKLIB_NAME} PUBLIC rdkit_base)
     if(RDK_INSTALL_DEV_COMPONENT)
       INSTALL(TARGETS ${RDKLIB_NAME} EXPORT ${RDKit_EXPORTED_TARGETS}
               DESTINATION ${RDKit_LibDir}/${RDKLIB_DEST}
               COMPONENT dev )
     endif(RDK_INSTALL_DEV_COMPONENT)
-  else(MSVC)
+  else()
     # we're going to always build in shared mode since we
     # need exceptions to be (correctly) catchable across
     # boundaries. As of now (June 2010), this doesn't work
     # with g++ unless libraries are shared.
-      add_library(${RDKLIB_NAME} SHARED ${RDKLIB_SOURCES})
-      target_link_libraries(${RDKLIB_NAME} PUBLIC rdkit_base)
-      INSTALL(TARGETS ${RDKLIB_NAME} EXPORT ${RDKit_EXPORTED_TARGETS}
-              DESTINATION ${RDKit_LibDir}/${RDKLIB_DEST}
-              COMPONENT runtime )
-      if(RDK_INSTALL_STATIC_LIBS)
-        add_library(${RDKLIB_NAME}_static ${RDKLIB_SOURCES})
+    add_library(${RDKLIB_NAME} SHARED ${RDKLIB_SOURCES})
+    target_link_libraries(${RDKLIB_NAME} PUBLIC rdkit_base)
+    INSTALL(TARGETS ${RDKLIB_NAME} EXPORT ${RDKit_EXPORTED_TARGETS}
+            DESTINATION ${RDKit_LibDir}/${RDKLIB_DEST}
+            COMPONENT runtime )
+    if(RDK_INSTALL_STATIC_LIBS)
+      add_library(${RDKLIB_NAME}_static ${RDKLIB_SOURCES})
 
-        foreach(linkLib ${RDKLIB_LINK_LIBRARIES})
-          if(${linkLib} MATCHES "^(Boost)|(Thread)")
-            set(rdk_static_link_libraries "${rdk_static_link_libraries}${linkLib};")
-          else()
-            set(rdk_static_link_libraries "${rdk_static_link_libraries}${linkLib}_static;")
-          endif()
-        endforeach()
-        target_link_libraries(${RDKLIB_NAME}_static PUBLIC ${rdk_static_link_libraries})
-        target_link_libraries(${RDKLIB_NAME}_static PUBLIC rdkit_base)
-        if(RDK_INSTALL_DEV_COMPONENT)
-          INSTALL(TARGETS ${RDKLIB_NAME}_static EXPORT ${RDKit_EXPORTED_TARGETS}
+      foreach(linkLib ${RDKLIB_LINK_LIBRARIES})
+        if(${linkLib} MATCHES "^(Boost)|(Thread)")
```

Comment:
nothing to be done about it right now, but this is also going to have a conflict once #1858 is merged

Reply:
OK, I can wait for #1858 to be merged and solve the conflict myself.